Services/LTI/classes/Screen/LtiViewLayoutProvider.php

```diff
@@ -18,75 +27,116 @@
+                    $page = $this->getPage($parts);
+                    if ($isMemberView) {
+                        return $page->withModeInfo($this->dic->ui()->factory()->mainControls()->modeInfo($this->dic->language()->txt('mem_view_long'), $url));
+                    }
+                    else {
+                        return $page;
+                    }
+                }
+            );
         }
+        return null;
     }
 
-     /**
-     * This is a basic wip implemantion which is hiding the complete navigation MainBar. 
-     * We are planning to create an own LTI MainBar with a minimal TopItem Entry p.e. "LTI Home" with slate like this (needs to be discussed):
-     *  
-     * LtiRoot Object
-     * -- Separator --
-     * LastVisited Items
-     * -- Separator --
-     * Delete VistiedItems (?)
-     */
-     
-    public function getMainBarModification(CalledContexts $screen_context_stack) : ?MainBarModification
-    {
-        if ($this->dic["lti"]->isActive()) {
-            return $this->globalScreen()
-                ->layout()
-                ->factory()
-                ->mainbar()
-                ->withModification(function (MainBar $current) : ?MainBar { return null; })->withHighPriority();
-        }
-        else {
-            return null;
-        }
+    private function getTitle(): String {
+       return $this->dic["lti"]->getTitleBar(true); 
+    }
+    
+    private function getShortTitle() : String {
+       return ($this->dic["lti"]->getShortTitle()) ? $this->dic["lti"]->getShortTitle() : "";
+    }
+    
+    private function getViewTitle() : String {
+       return ($this->dic["lti"]->getViewTitle()) ? $this->dic["lti"]->getViewTitle() : "";
+    }
+    
+    private function getMetaBar() : MetaBar {
+        $f = $this->dic->ui()->factory();
+        $close = $f->button()->close();
+        $exit_symbol = $f->symbol()->glyph()->remove();
+        $exit_txt = $this->dic['lti']->lng->txt('lti_exit');
+        $exit = $f->button()->bulky($exit_symbol,$exit_txt,$this->dic["lti"]->getCmdLink('exit'));
+        $metabar = $f->mainControls()->metaBar()->withAdditionalEntry('exit', $exit);
+        return $metabar;
+    }
+
+    private function getMainBar() : Mainbar {
+        $if = $this->globalScreen()->identification()->core($this);
+        $f = $this->dic->ui()->factory();
+        $mb = $f->mainControls()->mainBar();
+        $title = ($this->dic["lti"]->getHomeTitle() != "") ? $this->dic["lti"]->getHomeTitle() : "LTI Home";
+        $link = ($this->dic["lti"]->getHomeLink() != "") ? $this->dic["lti"]->getHomeLink() : "#";
+        $icon = $this->dic->ui()->factory()->symbol()->icon()->custom(\ilUtil::getImagePath("simpleline/home.svg"), $title);
+        $lti_home = $this->globalScreen()->mainbar()->topLinkItem($if->identifier('mm_lti_home'))
+            ->withSymbol($icon)
+            ->withTitle($title)
+            ->withAction($link);
+        $renderer = new TopLinkItemRenderer();
```

Comment:
Can you post a screenshot of what the whole thing looks like (or should look like)? Maybe we can do it a lot easier than that

Reply:
OK..
I will post some screenshots with different resolutions, because i was really impressed of the responsivness by the straight usage of the new UI components, so this LTI provider will work on a smartphone too, , thx!!